AppSettingsManagerAPI/AppSettingsManagerApi.Infrastructure.Tests/MySqlPermissionRepositoryTests.cs

```diff
@@ -26,6 +32,410 @@ public MySqlPermissionRepositoryTests()
+
+        var convertedPermission = _fixture.Build<Model.Permission>().Create();
+
+        _permissionConverterMock
+            .Setup(
+                s =>
+                    s.Convert(
+                        It.Is<Permission>(
+                            p =>
+                                p.SettingGroupId == settingGroup.Id
+                                && p.UserId == newUser.Id
+                                && p.CurrentPermissionLevel == PermissionLevel.None
+                                && p.RequestedPermissionLevel == PermissionLevel.None
+                                && p.WaitingForApproval == false
+                                && p.ApprovedBy == string.Empty
+                        )
+                    )
+            )
+            .Returns(convertedPermission);
+
+        response.Approved = false;
+        response.SettingGroupId = settingGroup.Id;
+        response.ApproverId = userAdmin.Id;
+        response.UserId = newUser.Id;
+        response.Password = userAdmin.Password;
+
+        var actualPermission = await _permissionRepository.PermissionRequestResponse(response);
+
+        Assert.Equal(convertedPermission, actualPermission);
+    }
+
+    [Theory]
+    [AutoTestData]
+    public async Task DeletePermission_Succeeds(string userId, string settingGroupId)
+    {
+        var user = _fixture.Build<User>().Create();
+        user.Id = userId;
+        user.Permissions = new List<Permission>();
+        _settingsContext.Users.Add(user);
+
+        var settingGroup = _fixture.Build<SettingGroup>().Create();
+        settingGroup.Id = settingGroupId;
+        settingGroup.Permissions = new List<Permission>();
+        _settingsContext.SettingGroups.Add(settingGroup);
+
+        var permission = _fixture.Build<Permission>().Create();
+        permission.UserId = userId;
+        permission.User = user;
+        permission.SettingGroupId = settingGroupId;
+        permission.SettingGroup = settingGroup;
+
+        _settingsContext.Permissions.Add(permission);
+        await _settingsContext.SaveChangesAsync();
+
+        var convertedPermission = _fixture.Build<Model.Permission>().Create();
+
+        _permissionConverterMock
+            .Setup(
+                s =>
+                    s.Convert(
+                        It.Is<Permission>(
+                            p => p.SettingGroupId == settingGroup.Id && p.UserId == user.Id
+                        )
+                    )
+            )
+            .Returns(convertedPermission);
+
+        var actualPermission = await _permissionRepository.DeletePermission(userId, settingGroupId);
+
+        Assert.Equal(convertedPermission, actualPermission);
+        await Assert.ThrowsAsync<InvalidOperationException>(
+            () =>
+                _settingsContext.Permissions.SingleAsync(
+                    p => p.SettingGroupId == settingGroupId && p.UserId == userId
+                )
+        );
+    }
+
```

Comment:
Looks like you made some changes to the unit tests I wrote, and it looks even better now. Great stuff!